block_my_feedback.php

```diff
@@ -59,35 +61,220 @@ public function get_content(): stdClass {
+        // User courses.
+        $courses = enrol_get_all_users_courses($user->id, false, ['enddate']);
+        // Marking.
+        $marking = [];
+
+        foreach ($courses as $course) {
+            // Skip hidden courses.
+            if (!$course->visible) {
+                continue;
+            }
+            // Skip none current course.
+            if (!self::is_course_current($course)) {
+                continue;
+            }
+            // Skip if no summative assessments.
+            if (!$summatives = assess_type::get_assess_type_records_by_courseid($course->id, 1)) {
+                continue;
+            }
+
+            $modinfo = get_fast_modinfo($course->id);
+
+            // Loop through assessments for this course.
+            foreach ($summatives as $summative) {
+
+                // Check this is a course mod.
+                if (isset($summative->cmid)) {
+                    $cmid = $summative->cmid;
+                    $mod = $modinfo->get_cm($cmid);
+
+                    // Skip hidden mods.
+                    if (!$mod->visible) {
+                        continue;
+                    }
+
+                    // Template.
+                    $assess = new stdClass;
+                    $assess->cmid = $cmid;
+                    $assess->modname = $mod->modname;
+                    // Get due date and require marking.
+                    $assess = self::get_mod_data($mod, $assess);
+
+                    // Check mod has require marking (only set when there is a due date).
+                    if (isset($assess->requiremarking)) {
+                        // TODO - what is expensive here that we can do after sort and limit?
+                        $assess->name = $mod->name;
+                        $assess->coursename = $course->fullname;
+                        $assess->url = new moodle_url('/mod/'. $mod->modname. '/view.php', ['id' => $cmid]);
+                        $assess->icon = course_summary_exporter::get_course_image($course);
+                        $marking[] = $assess;
+                    }
+                }
+            }
+        }
+
+        // Sort and return data.
+        if ($marking) {
+            usort($marking, function ($a, $b) {
+                return $a->unixtimestamp <=> $b->unixtimestamp;
+            });
+
+            return array_slice($marking, 0, 5);
+        }
+        return null;
+    }
+
+    /**
+     * Return mod data - due date & require marking.
+     *
+     * TODO - turnitin, quiz.
+     *
+     * @param stdClass $mod
+     * @param stdClass $assess
+     */
+    public function get_mod_data($mod, $assess) {
+        global $CFG;
+        // Mods have different feilds for due date, and require marking.
+        switch ($mod->modname) {
+            case 'assign':
+
+                // Check mod due date is relevant.
+                $duedate = self::duedate_in_range($mod->customdata['duedate']);
```

Comment:
`duedate_in_range` is not a static function, call it by using $`this` or change the method to static?